projects/igniteui-angular/src/lib/grids/grid/grid.component.ts

```diff
@@ -826,7 +826,11 @@ export class IgxGridComponent extends IgxGridBaseDirective implements GridType,
     /**
     * @hidden @internal
     */
-    public getContext(rowData, rowIndex): any {
+    public getContext(rowData: any, rowIndex: number, pinned?: boolean): any {
+        if (pinned && !this.isRowPinningToTop) {
```

Comment:
@MayaKirova @ChronosSF This should definitely be noted in the feature documentation, that pinned rows will affect the start index of normal rows or are affected by the length of the data view when on the bottom.